velox/expression/tests/CustomTypeTest.cpp

```diff
@@ -109,13 +125,36 @@ class FromFancyIntFunction : public exec::VectorFunction {
                 .build()};
   }
 };
+
+struct FancyIntT {
+  using type = std::shared_ptr<FancyInt>;
+  static constexpr const char* typeName = "fancy_int";
+};
+using TheFancyInt = CustomType<FancyIntT>;
```

Comment:
It looks a bit confusing that we have to create 2 types one derived from CustomType and another from OpaqueType. I would imagine an OpaqueType should be a CustomType. Please can you give some info about it.

Reply:
@aditi-pandit This is explained in #2965. Also, see #2986 and #2969 .